File: cellulosesz-api/src/main/java/top/likoslupus/cellulosesz/api/platform/CellPlayer.java
package top.likoslupus.cellulosesz.api.platform;

import java.util.UUID;

import static top.likoslupus.cellulosesz.api.validation.TextChecks.requireNonBlank;

import static java.util.Objects.requireNonNull;

/** Stable platform-neutral player identity. */
public record CellPlayer(
        UUID uuid,
        String name
) {

    public CellPlayer {
        requireNonNull(uuid, "uuid");
        name = requireNonBlank(requireNonNull(name, "name").trim(), "name");
    }

}

File: cellulosesz-api/src/main/kotlin/top/likoslupus/cellulosesz/api/platform/CellPlayer.kt
package top.likoslupus.cellulosesz.api.platform

import top.likoslupus.cellulosesz.api.validation.requireNonBlank
import java.util.*

/** Stable platform-neutral player identity. */
@JvmRecord
public data class CellPlayer(
    public val uuid: UUID,
    public val name: String
) {

    init {
        name.requireNonBlank { "name" }
    }

}
